src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -793,6 +796,26 @@ def update_children(event: TraceEvent, limit: int) -> None:
                 child.generation = parent.generation + 1 if parent.generation is not None else None
                 parents.append(child)
 
+    # Concurrently fetches nodestore data to construct and return a dict mapping eventid of a txn
+    # to the associated nodestore event.
+    @staticmethod
+    def nodestore_event_map(events: Sequence[SnubaTransaction]) -> Dict[str, Optional[Event]]:
+        map = {}
+        with ThreadPoolExecutor(max_workers=20) as executor:
```

Comment:
This should be feature flagged, right?

Reply:
I also don't really see us going above 10 in the backend, maybe we should use that limit here as well?

Reply:
Yep you're right I'll feature flag it